### DISCUSSION TITLE
Vercel build uses preview ENV variables when merging into production

### DISCUSSION THREAD

**Discussion by davidfloegel** [vercel/community#4916] [Help] [answered]
### Summary

Hi there,

I've started experiencing a rather odd behaviour that I've not seen before.

In my vercel app, I've got a production branch (main) as well as a staging branch (dev). I have separated my env variables by using the `production` and `preview` env variables.

Now, when I deploy to staging and test that everything is fine, it uses the correct (preview) environment variables. However, when I create a PR, and merge my `dev` branch into `main`, it deploys _still_ using the preview environment variables, instead of using production. I've made sure that my `main` branch is set to be the `production` branch in my Git settings within Vercel. 

This is pretty annoying, because not only do my users then see a staging environment on production, but I have to manually click "redeploy" for it to rebuild with the prod env variables.

I've not had this happen before on any of the apps I've worked on. Am I missing something super obvious?

Kind regards
David

### Steps to Reproduce

No error messages.

**Comment by amyegan** [comment]
Does your project use Turborepo or Nx?

**Reply by davidfloegel** [reply]
NX - sorry I should've mentioned. 

However, the project we have at work also uses NX and we don't have that issue there 🤔

**Reply by amyegan** [reply, marked as answer]
No worries! It's possible that this is a remote cache problem. The Nx docs have a [How Caching Works](https://nx.dev/concepts/how-caching-works#runtime-hash-inputs) page. That should help if caching the source of the problem.

If that's not it, I could use a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) to dig further into the problem.

**Reply by davidfloegel** [reply]
Brilliant, thanks @amyegan - I've just disabled caching for now, that solved it!
